knime_extension/src/nodes/transform.py

```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
+        is_advanced=True,
+    )
+
+    def configure(self, configure_context, input_schema):
+        self.o_geo_col = knut.column_exists_or_preset(
+            configure_context, self.o_geo_col, input_schema, knut.is_geo
+        )
+        self.d_geo_col = knut.column_exists_or_preset(
+            configure_context, self.d_geo_col, input_schema, knut.is_geo
+        )
+
+        return self.result_settings.get_result_schema(
+            configure_context,
+            input_schema,
+            "Curve",
+            knut.TYPE_LINE,
+        )
+
+    def execute(self, exec_context: knext.ExecutionContext, input_1):
+        gdf = knut.load_geo_data_frame(input_1, self.o_geo_col, exec_context)
+
+        import numpy as np
+        from shapely.geometry import LineString
+
+        def create_line_geometry(row):
+            p0 = row[self.o_geo_col].coords[0]
+            p2 = row[self.d_geo_col].coords[0]
+            return LineString([p0, p2])
+
+        def bezier_curve(p0, cp, p2, num_points=100):
+            t_values = np.linspace(0, 1, num_points)
+            curve = (
+                np.outer((1 - t_values) ** 2, p0)
+                + np.outer(2 * (1 - t_values) * t_values, cp)
+                + np.outer(t_values**2, p2)
+            )
+            return curve
+
+        def calculate_control_point(p0, p2, height_scale=0.3, angle_degrees=60):
+            midpoint = (p0 + p2) / 2
+            distance = np.linalg.norm(p2 - p0)
+            offset_distance = distance * height_scale
+            vec = p2 - p0
+            angle_radians = np.radians(angle_degrees)
+            rotation_matrix = np.array(
+                [
+                    [np.cos(angle_radians), np.sin(angle_radians)],
+                    [-np.sin(angle_radians), np.cos(angle_radians)],
+                ]
+            )
+            vec_perpendicular = np.dot(rotation_matrix, vec)
+            vec_perpendicular /= np.linalg.norm(vec_perpendicular)
+            vec_perpendicular *= offset_distance
+            control_point = midpoint + vec_perpendicular
+            return control_point
+
+        def create_bezier_line_geometry(
+            row,
+            height_scale=self.height_scale,
+            angle_degrees=self.angle_degrees,
+            num_points=self.num_points,
+        ):
+            p0 = np.array(row[self.o_geo_col].coords[0])
+            p2 = np.array(row[self.d_geo_col].coords[0])
+
+            # Calculate the control point
+            cp = calculate_control_point(
+                p0, p2, height_scale=self.height_scale, angle_degrees=self.angle_degrees
+            )
+
+            # Calculate the points on the Bezier curve
+            curve_points = bezier_curve(p0, cp, p2, num_points=self.num_points)
+
+            # Convert the curve points to a LineString geometry
+            line_geom = LineString(curve_points)
+
+            return line_geom
+
+        if self.result_settings.mode == knut.ResultSettingsMode.APPEND.name:
```

Comment:
I would simply always append a new column to the input table with name Curve. Just make sure that the name is unique by using the `knut.get_unique_column_name()` method

Reply:
revised